generator/routing_helpers.cpp

```diff
@@ -10,41 +10,73 @@
 using std::map;
 using std::string;
 
-namespace routing
-{
-void AddFeatureId(map<uint64_t, uint32_t> & osmIdToFeatureId, uint32_t featureId, uint64_t osmId)
+namespace
 {
-  auto const result = osmIdToFeatureId.insert(make_pair(osmId, featureId));
-  if (!result.second)
-  {
-    LOG(LERROR, ("Osm id", osmId, "is included in two feature ids:", featureId,
-                 osmIdToFeatureId.find(osmId)->second));
-  }
-}
-
-bool ParseOsmIdToFeatureIdMapping(string const & osmIdsToFeatureIdPath,
-                                  map<uint64_t, uint32_t> & osmIdToFeatureId)
+template <class ToDo>
+bool ForEachRoadFromFile(string const & filename, ToDo && toDo)
 {
   gen::OsmID2FeatureID osmIdsToFeatureIds;
   try
   {
-    FileReader reader(osmIdsToFeatureIdPath);
+    FileReader reader(filename);
     ReaderSource<FileReader> src(reader);
     osmIdsToFeatureIds.Read(src);
   }
   catch (FileReader::Exception const & e)
   {
-    LOG(LWARNING, ("Exception while reading file:", osmIdsToFeatureIdPath, ". Msg:", e.Msg()));
+    LOG(LERROR, ("Exception while reading file:", filename, ". Msg:", e.Msg()));
     return false;
   }
 
   osmIdsToFeatureIds.ForEach([&](gen::OsmID2FeatureID::ValueT const & p) {
     if (p.first.IsWay())
-    {
-      AddFeatureId(osmIdToFeatureId, p.second /* feature id */, p.first.OsmId() /* osm id */);
-    }
+      toDo(p.second /* feature id */, p.first.OsmId());
   });
 
   return true;
 }
+}  // namespace
+
+namespace routing
+{
+void AddFeatureId(map<uint64_t, uint32_t> & osmIdToFeatureId, uint32_t featureId, uint64_t osmId)
+{
+  auto const result = osmIdToFeatureId.insert(make_pair(osmId, featureId));
+  if (!result.second)
+  {
+    LOG(LERROR, ("Osm id", osmId, "is included in two feature ids:", featureId,
+                 osmIdToFeatureId.find(osmId)->second));
+  }
+}
+
+bool ParseOsmIdToFeatureIdMapping(string const & osmIdsToFeatureIdPath,
+                                  map<uint64_t, uint32_t> & osmIdToFeatureId)
+{
+  return ForEachRoadFromFile(osmIdsToFeatureIdPath, [&](uint32_t featureId, uint64_t osmId) {
+    AddFeatureId(osmIdToFeatureId, featureId, osmId);
+  });
+}
+
+bool ParseFeatureIdToOsmIdMapping(string const & osmIdsToFeatureIdPath,
+                                  map<uint32_t, uint64_t> & featureIdToOsmId)
+{
+  bool result = true;
+
+  result &= ForEachRoadFromFile(osmIdsToFeatureIdPath, [&](uint32_t featureId, uint64_t osmId) {
```

Comment:
Please, don't mix assignment with modification in the ForEachRoadFromFile. For readability, use two flags.